src/fidesops/ops/service/pagination/pagination_strategy_link.py

```diff
@@ -10,24 +10,25 @@
 from fidesops.ops.schemas.saas.strategy_configuration import (
     LinkPaginationConfiguration,
     LinkSource,
-    StrategyConfiguration,
 )
 from fidesops.ops.service.pagination.pagination_strategy import PaginationStrategy
+from fidesops.ops.service.strategy_factory import register
 from fidesops.ops.util.logger import Pii
 
-STRATEGY_NAME = "link"
-
 logger = logging.getLogger(__name__)
 
 
+@register
 class LinkPaginationStrategy(PaginationStrategy):
+
+    name = "link"
+    configuration_model = LinkPaginationConfiguration
+
     def __init__(self, configuration: LinkPaginationConfiguration):
         self.source = configuration.source
         self.rel = configuration.rel
         self.path = configuration.path
-
-    def get_strategy_name(self) -> str:
-        return STRATEGY_NAME
+        super().__init__(configuration)
```

Comment:
Generally, it feels like we're trying to force all of these disparate strategies to all be registered in the same way, when they are distinct types of objects.  Their individual configurations are hardly related and I think it would cause confusion for someone trying to create their own strategy.

I might recommend keeping them separate, and allowing a user to creating their own "masking strategy", "pagination strategy", "post processor strategy", or "authentication strategy".